pkg/lsp/service.go

```diff
@@ -38,8 +38,20 @@ const (
 	TDSKIncremental                      = 2
 )
 
+type TextDocumentSyncOptions struct {
+	OpenClose         bool                 `json:"openClose,omitempty"`
+	Change            TextDocumentSyncKind `json:"change"`
+	WillSave          bool                 `json:"willSave,omitempty"`
+	WillSaveWaitUntil bool                 `json:"willSaveWaitUntil,omitempty"`
+	Save              *SaveOptions         `json:"save,omitempty"`
+}
+
+type SaveOptions struct {
+	IncludeText bool `json:"includeText"`
+}
+
 type ServerCapabilities struct {
-	TextDocumentSync                 int                              `json:"textDocumentSync,omitempty"`
+	TextDocumentSync                 TextDocumentSyncOptions          `json:"textDocumentSync,omitempty"`
```

Comment:
don't know this off the top of my head, but how does go handle unmarshalling something it expects to be an int, but it gets `TextDocumentSyncOptions` instead? Unless we do some sort of atomic rollout of this, that is gonna be an issue unless it just silently fails.